internal/service/support/support_case.go

```diff
@@ -0,0 +1,261 @@
+	var plan resourceSupportCaseData
+	response.Diagnostics.Append(request.Plan.Get(ctx, &plan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().SupportClient(ctx)
+
+	input := support.CreateCaseInput{
+		CommunicationBody: flex.StringFromFramework(ctx, plan.CommunicationBody),
+		Subject:           flex.StringFromFramework(ctx, plan.Subject),
+		CategoryCode:      flex.StringFromFramework(ctx, plan.CategoryCode),
+		CcEmailAddresses:  flex.ExpandFrameworkStringValueList(ctx, plan.CCEmailAddresses),
+		IssueType:         flex.StringFromFramework(ctx, plan.IssueType),
+		Language:          flex.StringFromFramework(ctx, plan.Language),
+		ServiceCode:       flex.StringFromFramework(ctx, plan.ServiceCode),
+		SeverityCode:      flex.StringFromFramework(ctx, plan.SeverityCode),
+	}
+	output, err := conn.CreateCase(ctx, &input)
+	if err != nil {
+		response.Diagnostics.Append(create.DiagErrorFramework(names.Support, create.ErrActionCreating, ResourceNameSupportCase, plan.Subject.String(), err))
+		return
+	}
+
+	// Create API call returns only Case ID. Get other details as well.
+	caseDetails, err := findSupportCaseByID(ctx, conn, *output.CaseId)
+	if err != nil {
+		response.Diagnostics.Append(create.DiagErrorFramework(names.Support, create.ErrActionChecking, ResourceNameSupportCase, plan.Subject.String(), err))
+		return
+	}
+
+	state := plan
+	state.refreshFromOutput(ctx, caseDetails)
+
+	response.Diagnostics.Append(response.State.Set(ctx, state)...)
+}
+
+func (r *resourceSupportCase) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state resourceSupportCaseData
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().SupportClient(ctx)
+
+	caseDetails, err := findSupportCaseByID(ctx, conn, state.CaseID.ValueString())
+	if tfresource.NotFound(err) {
+		create.LogNotFoundRemoveState(names.Support, create.ErrActionReading, ResourceNameSupportCase, state.ID.ValueString())
+		response.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		response.Diagnostics.Append(create.DiagErrorFramework(names.Support, create.ErrActionReading, ResourceNameSupportCase, state.ID.String(), err))
+		return
+	}
+
+	state.refreshFromOutput(ctx, caseDetails)
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+// Update is a no-op, for now. For another no-op update example, see service/auditmanager/organization_admin_account_registration.go
+func (r *resourceSupportCase) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+}
+
+// Delete is a no-op, because AWS doesn't provide a deletion API. For another no-op update example, see service/auditmanager/organization_admin_account_registration.go
+func (r *resourceSupportCase) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
```

Comment:
```suggestion
// Delete is a no-op, because AWS doesn't provide a deletion API. For another no-op delete example, see service/auditmanager/organization_admin_account_registration.go
func (r *resourceSupportCase) Delete(_ context.Context, _ resource.DeleteRequest, _ *resource.DeleteResponse) {
```

Reply:
Done.